packages/terra-worklist-data-grid/src/subcomponents/Row.jsx

```diff
@@ -0,0 +1,138 @@
+   * rendered to allow for row selection to occur.
+   */
+  hasSelectableRows: PropTypes.bool,
+  /**
+   * Callback function that will be called when a row is selected. Parameters: `onRowSelect(rowId)`
+   */
+  onRowSelect: PropTypes.func,
+  /**
+   * All columns currently displayed.
+   */
+  displayedColumns: PropTypes.arrayOf(WorklistDataGridPropTypes.columnShape),
+  /**
+   * Callback function that will be called when a cell in the row is selected.
+   */
+  onCellSelect: PropTypes.func,
+  /**
+   * A number indicating which column represents the row header. This is zero based.
+   */
+  rowHeaderIndex: PropTypes.number,
+  /**
+   * Id of the column in the row that is selected.
+   */
+  selectedCellColumnId: PropTypes.string,
+  /**
+   * Column index of the column in that has focus.
+   */
+  tabStopColumnIndex: PropTypes.number,
+
+};
+
+const defaultProps = {
+  hasSelectableRows: false,
+};
+
+function Row(props) {
+  const {
+    rowIndex,
+    height,
+    hasSelectableRows,
+    row,
+    displayedColumns,
+    rowHeaderIndex,
+    onRowSelect,
+    onCellSelect,
+    selectedCellColumnId,
+    tabStopColumnIndex,
+  } = props;
+
+  const theme = useContext(ThemeContext);
+
+  const handleCellSelect = (rowIdColId, coordinates) => {
+    if (hasSelectableRows) {
+      if (onRowSelect) {
+        onRowSelect(rowIdColId.rowId, rowIndex, coordinates);
+      }
+    } else if (onCellSelect) {
+      onCellSelect(rowIdColId, coordinates);
+    }
+  };
+
+  const getCellData = (cellRowIndex, cellColumnIndex, cell, rowId) => {
+    const columnId = displayedColumns[cellColumnIndex].id;
+    const isRowHeader = cellColumnIndex === (rowHeaderIndex + hasSelectableRows ? 1 : 0);
+
+    return (
+      <Cell
+        rowId={rowId}
+        columnId={columnId}
+        rowIndex={cellRowIndex}
+        columnIndex={cellColumnIndex}
+        key={`${rowId}_${columnId}`}
+        isTabStop={tabStopColumnIndex === cellColumnIndex}
+        isSelected={!hasSelectableRows && selectedCellColumnId === columnId}
+        cell={cell}
+        isRowHeader={isRowHeader}
+        onCellSelect={handleCellSelect}
+      />
+    );
+  };
+
+  const columnIndexOffSet = hasSelectableRows ? 1 : 0;
+  return (
+    <tr
+      className={cx([row.isSelected ? 'row-selected' : 'worklist-data-grid-row', theme.className])}
+      // eslint-disable-next-line react/forbid-dom-props
+      style={{ height }}
+    >
+      {hasSelectableRows && (
+        <RowSelectionCell
+          rowId={row.id}
+          columnId={displayedColumns[0].id}
+          rowIndex={rowIndex}
+          columnIndex={0}
+          isTabStop={tabStopColumnIndex === 0}
+          isSelected={row.isSelected}
+          ariaLabel={row.ariaLabel}
+          onCellSelect={handleCellSelect}
+        />
+      )}
+      {row.cells.map((cell, cellColumnIndex) => (
+        getCellData(rowIndex, cellColumnIndex + columnIndexOffSet, cell, row.id)
+      ))}
+    </tr>
+  );
+}
```

Comment:
Can we update this to use a ternary for better readability?
```suggestion
const rowHeader = hasSelectableRows ? 
  <RowSelectionCell
  rowId={row.id}
  columnId={displayedColumns[0].id}
  rowIndex={rowIndex}
  columnIndex={0}
  isTabStop={tabStopColumnIndex === 0}
  isSelected={row.isSelected}
  ariaLabel={row.ariaLabel}
  onCellSelect={handleCellSelect}
  /> :
  null;

return (
  <tr
    className={cx([row.isSelected ? 'row-selected' : 'worklist-data-grid-row', theme.className])}
    // eslint-disable-next-line react/forbid-dom-props
    style={{ height }}
  >
    {rowHeader}
    {row.cells.map((cell, cellColumnIndex) => (
      getCellData(rowIndex, cellColumnIndex + columnIndexOffSet, cell, row.id)
    ))}
  </tr>
);
```

Reply:
Done.